### PR TITLE
Refactored User model and API

### LINKED ISSUE #3795: Enhance user model/api
**Rename fields:**

- [ ] `avatar` -> `avatar_url`
- [ ] `firstname` -> `first_name`
- [ ] `lastname` -> `last_name`
- [ ] `facebook` -> `facebook_url`
- [ ] `twitter` -> `twitter_url`
- [ ] `instagram` -> `instagram_url`
- [ ] `google` -> `google_plus_url`
- [ ] `thumbnail` -> `thumbnail_image_url`
- [ ] `small` -> `small_image_url`
- [ ] `icon` -> `icon_image_url`
- [ ] `avatar` -> `original_image_url`

**Remove fields:**

- [ ] `avatar_uploaded` -> remove. no use. 
- [ ] `signup_at` -> remove. `created_at` provides the same data.

**Handle image resizing and save:**

When ever `original_image_url` changes (ignore any changes to `small_image_url`, `thumbnail_image_url`, `icon_image_url`), download, resize it appropriately, upload it back to the storage and update `small_image_url`, `thumbnail_image_url`, `icon_image_url` fields with the new URLs

### PR DESCRIPTION
**Pull request by SaptakS**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 

#### Short description of what this resolves:
Refactored column names and attributes for users

Fixes #3795

### CODE REVIEW

File: app/api/users.py [hunk 1]
```diff
@@ -31,42 +29,42 @@ class Meta:
...
-                                 related_view='v1.event_invoice_list',
-                                 related_view_kwargs={'user_id': '<id>'},
-                                 schema='EventInvoiceSchema',
-                                 many=True,
-                                 type_='event-invoice')
+    facebook_url = fields.Url()
+    twitter_url = fields.Url()
+    instagram_url = fields.Url()
+    google_plus_url = fields.Url()
+    thumbnail_image_url = fields.Url(attribute='thumbnail')
```
Comment (poush) [inline app/api/users.py]: You forgot to change this `(attribute='thumbnail')`

File: app/api/users.py [hunk 2]
```diff
@@ -31,42 +29,42 @@ class Meta:
...
+    small_image_url = fields.Url(attribute='small')
```
Comment (poush) [inline app/api/users.py]: here also

File: app/api/users.py [hunk 3]
```diff
@@ -31,42 +29,42 @@ class Meta:
...
+    icon_image_url = fields.Url(attribute='icon')
```
Comment (poush) [inline app/api/users.py]: and here

File: app/api/users.py (hunk 1 above)
Comment (SaptakS) [inline app/api/users.py]: Thanks for pointing out.

Comment (SaptakS) [inline app/api/users.py]: Updated.

**Pull request merged**